src/arkode/arkode_sprkstep_io.c

```diff
@@ -0,0 +1,491 @@
+}
+
+int SPRKStepSetAdaptivityFn(void* arkode_mem, ARKAdaptFn hfun, void* h_data)
+{
+  return (arkSetAdaptivityFn(arkode_mem, hfun, h_data));
+}
+
+int SPRKStepSetFixedStep(void* arkode_mem, realtype hfixed)
+{
+  return (arkSetFixedStep(arkode_mem, hfixed));
+}
+
+/*===============================================================
+  SPRKStep Optional output functions (wrappers for generic ARKODE
+  utility routines).  All are documented in arkode_io.c.
+  ===============================================================*/
+int SPRKStepGetNumStepAttempts(void* arkode_mem, long int* nstep_attempts)
+{
+  return (arkGetNumStepAttempts(arkode_mem, nstep_attempts));
+}
+
+int SPRKStepGetNumSteps(void* arkode_mem, long int* nsteps)
+{
+  return (arkGetNumSteps(arkode_mem, nsteps));
+}
+
+int SPRKStepGetLastStep(void* arkode_mem, realtype* hlast)
+{
+  return (arkGetLastStep(arkode_mem, hlast));
+}
+
+int SPRKStepGetCurrentStep(void* arkode_mem, realtype* hcur)
+{
+  return (arkGetCurrentStep(arkode_mem, hcur));
+}
+
+int SPRKStepGetCurrentTime(void* arkode_mem, realtype* tcur)
+{
+  return (arkGetCurrentTime(arkode_mem, tcur));
+}
+
+int SPRKStepGetCurrentState(void* arkode_mem, N_Vector* state)
+{
+  return (arkGetCurrentState(arkode_mem, state));
+}
+
+int SPRKStepGetRootInfo(void* arkode_mem, int* rootsfound)
+{
+  return (arkGetRootInfo(arkode_mem, rootsfound));
+}
+
+int SPRKStepGetStepStats(void* arkode_mem, long int* nsteps, realtype* hinused,
+                         realtype* hlast, realtype* hcur, realtype* tcur)
+{
+  return (arkGetStepStats(arkode_mem, nsteps, hinused, hlast, hcur, tcur));
+}
+
+int SPRKStepGetNumConstrFails(void* arkode_mem, long int* nconstrfails)
+{
+  return (arkGetNumConstrFails(arkode_mem, nconstrfails));
+}
+
+int SPRKStepGetUserData(void* arkode_mem, void** user_data)
+{
+  return (arkGetUserData(arkode_mem, user_data));
+}
+
+char* SPRKStepGetReturnFlagName(long int flag)
+{
+  return (arkGetReturnFlagName(flag));
+}
+
+/*===============================================================
+  SPRKStep optional input functions -- stepper-specific
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  SPRKStepSetUserData:
+
+  Wrapper for generic arkSetUserData and arkLSSetUserData
+  routines.
+  ---------------------------------------------------------------*/
+int SPRKStepSetUserData(void* arkode_mem, void* user_data)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetUserData", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* set user_data in ARKODE mem */
+  retval = arkSetUserData(arkode_mem, user_data);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  return (ARK_SUCCESS);
```

Comment:
I see nothing stepper-specific about this routine, so it could be simplified and moved to the preceding section.
```suggestion
  return (arkSetUserData(arkode_mem, user_data));
```